CONTRIBUTING.md

```diff
@@ -0,0 +1,277 @@
+# Contributing to Muffin OS
+
+Welcome to Muffin OS! This guide will help you get started with contributing to this hobby x86-64 operating system kernel written in Rust.
+
+## Project Overview
+
+**Muffin OS** is a bare-metal operating system kernel that boots using the Limine bootloader and runs on QEMU. The project consists of ~109 Rust source files organized into a kernel and userspace components.
+
+- **Language:** Rust (Nightly)
+- **Target:** x86_64-unknown-none
+- **Bootloader:** Limine v9.x
+- **Build System:** Cargo with custom build scripts
+
+## Architecture
```

Comment:
Since the details of the structure will change regularly (crates renamed, new crates added etc.), please focus on the structure more abstractly (kernel/crates are kernel components that can be tested on the host etc.).